src/main/java/bio/terra/workspace/service/workspace/flight/CreateWorkspaceStep.java

```diff
@@ -55,9 +58,14 @@ public StepResult doStep(FlightContext flightContext) throws RetryException {
   public StepResult undoStep(FlightContext flightContext) {
     FlightMap inputMap = flightContext.getInputParameters();
     FlightMap workingMap = flightContext.getWorkingMap();
-    if (workingMap.get(CREATE_WORKSPACE_COMPLETED_KEY, Boolean.class)) {
-      UUID workspaceId = inputMap.get(WorkspaceFlightMapKeys.WORKSPACE_ID, UUID.class);
+    UUID workspaceId = inputMap.get(WorkspaceFlightMapKeys.WORKSPACE_ID, UUID.class);
+    if (workingMap.get(CREATE_WORKSPACE_COMPLETED_KEY, Boolean.class) != null) {
       workspaceDao.deleteWorkspace(workspaceId);
+    } else {
+      logger.warn(
+          "Undoing a CreateWorkspaceStep that was not fully completed. This may have created an orphaned workspace "
+              + workspaceId.toString()
+              + " in WM's database.");
```

Comment:
I'd like to find a way to tie certain log messages into alerts - it would be nice to get an alert if this happens, so we can react. Maybe this is as simple as making this a `logger.error`? Or maybe we insert a magic string e.g. `[ALERT]` into the message? @ansingh7115 as our alert expert, do you have any ideas here?

Reply:
We can create log-based metric and then set an alert on it
https://cloud.google.com/logging/docs/logs-based-metrics/

Reply:
Sounds good - do we have an established pattern for what to put in the message contents for alerts (e.g. something like prefixing with `[ALERT]`?

Reply:
cool, so it sounds like no changes needed here!

Reply:
For my 2c, I think we should only log.error when we want to be alerted that something is *wrong*. I think the logging level of error vs warning vs info already gets us there and we shouldn't have special alert prefixing to follow on top of that. We will need to be disciplined about the logging level and demote things that aren't actually errors.
So some alerting for any error message (with silencing on a per line level) would be great.